substrate/frame/nfts/src/features/transfer.rs

```diff
@@ -231,4 +232,68 @@ impl<T: Config<I>, I: 'static> Pallet<T, I> {
 			Ok(())
 		})
 	}
+
+	/// Move the item and metadata deposits to the new owner.
+	///
+	/// - `collection`: The ID of the collection to which the NFT belongs.
+	/// - `item`: The ID of the NFT to move the deposit for.
+	/// - `new_owner`: The new owner's account.
+	/// - `collection_details`: Collection details
+	/// - `item_details`: Item details
+	///
+	/// This function moves the item and metadata deposits to the new owner.
+	/// If the deposit was placed by the collection owner, it will update the `owner_deposit`
+	/// accumulator on collection details.
+	///
+	/// Note: Attributes deposit is not moved.
+	pub(crate) fn do_move_item_deposit(
+		collection: T::CollectionId,
+		item: T::ItemId,
+		new_owner: T::AccountId,
+		collection_details: &mut CollectionDetailsFor<T, I>,
+		item_details: &mut ItemDetailsFor<T, I>,
+	) -> DispatchResult {
+		let ItemDepositOf::<T, I> { amount: item_deposit, account: item_depositor } =
+			item_details.deposit.clone();
+
+		if new_owner != item_depositor {
+			T::Currency::unreserve(&item_depositor, item_deposit);
+			T::Currency::reserve(&new_owner, item_deposit)?;
```

Comment:
Maybe it is common practice now, but this means that the caller must rollback the storage if it wants to handle the error.